prow/test/integration/test/webhook_server_test.go

```diff
@@ -123,3 +127,66 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
 		t.Errorf("prowjob was not mutated %v", err)
 	}
 }
+
+func TestWebhookServerAvailability(t *testing.T) {
+	clusterContext := getClusterContext()
+	kubeClient, err := NewClients("", clusterContext)
+	if err != nil {
+		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+	}
+	ctx := context.Background()
+	var podList corev1.PodList
+	webhookServerPod := &corev1.Pod{}
+	err = kubeClient.List(ctx, &podList, &client.ListOptions{})
+	if err != nil {
+		t.Fatalf("could not list pods %v", err)
+	}
+	for _, pod := range podList.Items {
+		if pod.ObjectMeta.Labels["app"] == "webhook-server" {
+			webhookServerPod = &pod
+			break
+		}
+	}
+	count := 0
+	for start := time.Now(); time.Since(start) < time.Second * 30; {
+		absProwJobPath, err := filepath.Abs(validBareProwJobPath)
+		if err != nil {
+			t.Fatalf("could not find absolute file path %v", err)
+		}
+		//read yaml file
+		prowJobYAML, err := ioutil.ReadFile(absProwJobPath)
+		if err != nil {
+			t.Fatalf("unable to read yaml file %v", err)
+		}
+		var prowJob v1.ProwJob
+		err = yaml.Unmarshal(prowJobYAML, &prowJob)
+		if err != nil {
+			t.Fatalf("unable to unmarshal into prowjob %v", err)
+		}
+		//update foo label of prowjob
+		val := prowJob.Labels["foo"]
+		prowJob.Labels["foo"] = val + strconv.Itoa(count)
```

Comment:
This appends the count rather than setting the value to the count. I would have expected this to quickly exceed the label length limits and cause errors so I'm surprised this is even passing TBH.